rot/cipher/DecryptionKeyGenerator.java

```diff
@@ -0,0 +1,21 @@
+package rot.cipher;
+
+public class DecryptionKeyGenerator extends CipherKeyGenerator{
+    @Override
+    public void generateKeyMap(int shift) {
+        EnglishAlphabetGenerator ABCGenerator = new EnglishAlphabetGenerator();
```

Comment:
Suggestion
public class DecryptionKeyGenerator extends CipherKeyGenerator {

    @Override
    public void generateKeyMap(int shift) {
        char[] alphabet = new EnglishAlphabetGenerator().getABC();
        int alphabetLength = alphabet.length;
        int adjustedShift = super.adjustShiftToCharsetLength(shift, alphabetLength);

        createCipherMapping(alphabet, adjustedShift);
    }

    private void createCipherMapping(char[] alphabet, int shift) {
        int alphabetLength = alphabet.length;
        for (int i = 0; i < alphabetLength; i++) {
            char keyChar = alphabet[i];
            char valueChar = alphabet[(i + shift) % alphabetLength];
            CipherKeyMap.put(valueChar, keyChar);
        }
    }
}